ibm/service/power/data_source_ibm_pi_datacenter.go

```diff
@@ -0,0 +1,94 @@
+// Copyright IBM Corp. 2017, 2021 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package power
+
+import (
+	"context"
+
+	"github.com/IBM-Cloud/power-go-client/clients/instance"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/flex"
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+const (
+	DatacenterRegion = "region"
+	DatacenterType   = "type"
+	DatacenterUrl    = "url"
+)
+
+func DataSourceIBMPIDatacenter() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceIBMPIDatacenterRead,
+		Schema: map[string]*schema.Schema{
+			Arg_DatacenterZone: {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.NoZeroValues,
+			},
+			Attr_DatacenterCapabilities: {
+				Type:        schema.TypeMap,
+				Computed:    true,
+				Description: "Datacenter Capabilities",
+				Elem: &schema.Schema{
+					Type: schema.TypeBool,
+				},
+			},
+			Attr_DatacenterHref: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Datacenter href",
+			},
+			Attr_DatacenterLocation: {
+				Type:        schema.TypeMap,
+				Computed:    true,
+				Description: "Datacenter location",
+			},
+			Attr_DatacenterStatus: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Datacenter status",
+			},
+			Attr_DatacenterType: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Datacenter type",
+			},
+		},
+	}
+}
+func dataSourceIBMPIDatacenterRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	// session
+	sess, err := meta.(conns.ClientSession).IBMPISession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	datacenterRegion := sess.Options.Zone
+	if region, ok := d.GetOk(Arg_DatacenterZone); ok {
+		datacenterRegion = region.(string)
```

Comment:
Zone=Region. It is confusing 👎

Reply:
Update to `datacenterZone` to clear confusion